include/range/v3/utility/any.hpp

```diff
@@ -61,7 +61,14 @@ namespace ranges
         template<typename T>
         T const * any_cast(any const *) noexcept;
 
+#if defined(RANGES_WORKAROUND_MSVC_589046) && !defined(RANGES_DOXYGEN_INVOKED)
+        namespace _any_ { struct adl_hook {}; }
+#endif
+
         struct any
+#if defined(RANGES_WORKAROUND_MSVC_589046) && !defined(RANGES_DOXYGEN_INVOKED)
+          : private _any_::adl_hook
+#endif
```

Comment:
empty base?

Reply:
*One* empty base class is fine, only classes with *multiple* potentially empty base classes need the `RANGES_EMPTY_BASES` annotation.